src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,253 @@
+#
+#  There are CI issues to be followed up for the test cases below that implements manually controlling sensor device for 
+#  the occupancy state ON/OFF change.
+#  [TC-OCC-3.1] test procedure step 4
+#  [TC-OCC-3.2] test precedure step 3c
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_OCC_2_1(MatterBaseTest):
+    async def read_occ_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.OccupancySensing
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_OCC_2_1(self) -> str:
+        return "[TC-OCC-2.1] Attributes with DUT as Server"
+
+    def steps_TC_OCC_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read Occupancy attribute."),
+            TestStep(3, "Read OccupancySensorType attribute."),
+            TestStep(4, "Read OccupancySensorTypeBitmap attribute."),
+            TestStep(5, "Read HoldTimeLimits attribute, if supported"),
+            TestStep(6, "Read HoldTime attribute, if supported"),
+            TestStep(7, "Read PIROccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(8, "Read PIRUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(9, "Read PIRUnoccupiedToOccupiedThreshold attribute, if supported"),
+            TestStep(10, "Read UltrasonicOccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(11, "Read UltrasonicUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(12, "Read UltrasonicUnoccupiedToOccupiedThreshold attribute, if supported"),
+            TestStep(13, "Read PhysicalContactOccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(14, "Read PhysicalContactUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(15, "Read PhysicalContactUnoccupiedToOccupiedThreshold attribute, if supported")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_1(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_1(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        self.step(1)
+        attributes = Clusters.OccupancySensing.Attributes
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(2)
+        if attributes.Occupancy.attribute_id in attribute_list:
+            occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+            asserts.assert_less_equal(occupancy_dut, 0b00000001, "Occupancy attribute is not in valid range")

```

Comment:
```suggestion
            asserts.assert_less_equal(occupancy_dut, 1, "Occupancy attribute is not in valid range")

```

Reply:
Occupancy attribute data type is bitmap, so I had it that way.